src/basics/LocalCache/ContextProvider.tsx

```diff
@@ -0,0 +1,67 @@
+// More info on Local storage: https://developer.mozilla.org/en-US/docs/Web/API/Window/localStorage
+import { log } from 'common/log';
+import * as React from 'react';
+import { useState, createContext, useContext, useCallback } from 'react';
+import { defaultLocalCacheConfig, LocalCacheItem } from './defaultConfig';
+
+export { LocalCacheItem } from './defaultConfig';
+
+interface LocalCacheState {
+  localCacheMap: Record<string, any>;
+  setLocalCache: (localCacheItem: LocalCacheItem, newValue: any) => void;
+  getLocalCache: (localCacheItem: LocalCacheItem) => any;
+}
+
+interface LocalCacheProviderProps {
+  children?: React.ReactNode;
+}
+
+// get the local storage value, if local storage is null, use default value
+function allStorage() {
+  const defaultKeys = Object.keys(defaultLocalCacheConfig);
+
+  var localCacheMap: Record<string, any> = {},
+    i = defaultKeys.length;
+  while (i--) {
+    const key = defaultKeys[i];
+    const value = localStorage.getItem(key) || defaultLocalCacheConfig[key];
+    localCacheMap[defaultKeys[i]] = JSON.parse(value);
+  }
+  return localCacheMap;
+}
+
+export const LocalCacheContext = createContext<LocalCacheState>({
+  localCacheMap: { ...allStorage() },
+  setLocalCache: () => {
+    /* Provider is not initialized */
+  },
+  getLocalCache: () => false,
+});
+
+export const LocalCacheProvider = (props: LocalCacheProviderProps) => {
```

Comment:
I'm curios why localCacheProvider was needed, did you get to many rerenders?

Reply:
since we want to share the Only Mine Setting State between NavBar and the checkmark in OnlyMyExecuations and other components. Therefore, we need to use a provider to wrap all children's components. Let me know if that doesn't make sense~

Reply:
`LocalCache` is stored in Browser Local Storage, which means it always has only one instance and always accessible for the code - even if we access it from different parts of the code or between app use.  You can find it here in DevTools:
<img width="613" alt="Screen Shot 2022-04-20 at 2 17 00 PM" src="https://user-images.githubusercontent.com/55718143/164324484-07cee8e6-c83a-41d9-964c-1c330e061260.png">

With this in mind CacheProvider shouldn't be needed to access the data, as previous useHook will read it from LocalStorage directly each time you request it. So in this case CacheProvider could be needed only in case if you think that we would have to many read/write operations so it will affect the performance. However, the reading/writing to LocalStorage from my experience is so inexpensive, that it doesn't really need to have a wrapper.

Reply:
Adding the Provider here - can even introduce issues, as now you additionally cache the value in the app. So if I will go to local storage and update the value, it won't be picked up on re-navigation.

Reply:
Oh I see why you did it. You needed a timed update, when you changed it one place, to notify the page that it was changed too.. I will think if we need to do anything around it, as I would like to avoid a Provider for all LocalStorage items here.

Reply:
yea. I need to get the update when OnlyMine changes the setting values. otherwise, can't update the checkbox in the dashboard. I see why you want to avoid a Provider for all LocalStorage items here. either we can find a solution to listen the localstorage change and update the state in Context or maybe we should create a provider just for navbar or general setting? 

(btw. normally I used redux for managing local storage, and redux will treat local storage as the normal state with the hook. so it won't have this issue..)

Reply:
btw. i think you change the value in localstorage and refresh. it should update for now.

Reply:
Yes, let's stick with provider. If you updated it manually it should be okay to require page refresh. 
You will need to update unit tests for it.
You can run tests per test file like that:
`yarn test LocalCache`